### ISSUE TITLE
Stop iSoQ from deleting explanation text when user selects a different assessment

### ISSUE THREAD

**Issue by MJWainwright** [epistemonikos/isoq#222]
A user has reported that it is quite annoying that all the text you type into the Explanation box is lost if you change your mind and select a different radio button. Can we please make it so that the text the user enters remains in the explanation box even if they change their selection in the radio buttons?
![image](https://github.com/epistemonikos/isoq/assets/72507850/a63bd9ce-fe34-42ce-be38-d7014a453602)

You can see this by doing the following:

- Click on assessment not completed for methological limitations
- select "minor concerns"
- type an explanation in the box
- change the selection to "moderate concerns" 
= the explanation disappears and the user has to retype it. we expect it to be fairly common that as a user writes out an explanation they may decide to change their rating in this way so it would be great if the text entered into the explanation box is saved.

**Comment by MJWainwright** [comment]
@damian-garrido Simon raised this as an issue again. I'm going to make it a priority

**Comment by MJWainwright** [comment]
@damian-garrido I'm going to check with Simon to see what he thinks as this was an issue he raised. I'm a little concerned that users will start free writing in that box rather than realising they have to complete the sentence. Then they will only realise this is what happens when they save and see that the text "minor concerns regarding..." has been added to the start of their text. But let me check with Simon and I'll get back to you soon. Thanks!

**Comment by damian-garrido** [comment]
@MJWainwright 
this will be not add text where the user write. this only show a text that previously was added in the box, but now out of the box.
the user will be write and this will not be erased, but the text in gray will be change if the user select other option.
there is no way to know if a user want to change the radio button without change the content on the text box.

**Comment by MJWainwright** [comment]
@damian-garrido . I got some feedback from Simon, Claire and another user about this. Firstly, we want the user to complete the sentence "X concerns regarding X because...". So we would want that text added automatically to whatever the user enters into the box once they have clicked save. However, we think that users may not know that this text will be added and might write an explanation which then doesn't make sense when "X concersn regarding X because..." is added to the start. Also, if it is added automatically then people cannot remove it, whereas right now they could simply delete the text "x concerns regarding X because" and write whatever they want. So the questions are:
1. can you think of any clever design ways to not have the "x concerns regarding X because..." in the box, but for it to be crystal clear to users that what they must do is complete that sentence? 
2. Is there any way we could allow users (but not encourage them to) edit the auto-added text "x concerns regarding x because..."

If there is no way around this then the feedback so far is that it is more annoying for the users to loose the explanation they enter when they change level of concern, then it would be to find out only after saving that "x concerns regarding x because..." will be added to to the start of their explanation.

**Comment by damian-garrido** [comment]
@MJWainwright what do you think if we explain that the gray content "X concerns regarding X because..." will be displayed on the final document, the exportable document, so, the user will be write in the textarea whatever they need, also, could change the selector, but when they export this, the gray text will be prepended to the text that the user introduce in the textarea.
for the moment, as i show you in the video, the gray text will be change with every changed selector.

**Comment by MJWainwright** [comment]
@damian-garrido, i don't think this solves the issue. We want that pre-text to appear not just in the exported document but in the online Evidence Profile. So whatever the user enters into the box, the text "minor concerns regarding methodological limitations because" is added to the start of the text and appears in the Evidence Profile table. The user will quickly learn that this text is added, and adjust accordingly. They will not be able to delete it, so we may get some complaints about that, but let's gather some user feedback. I think it's the best we can do for now. 

Please delete the sentence "Explain any concerns in your own words" and replace with "Select a level of concern above and complete the sentence that appears below to explain your concerns (not required for no/very minor concerns)":

Make the automated text "X concerns with X because" bold, remove the "...." at the end of the sentence and make the text size the same as the text size in the explanation box. 

![image](https://github.com/epistemonikos/isoq/assets/72507850/4931b00f-e52f-40e3-a4cd-75a6734bddc1)

**Comment by MJWainwright** [comment]
@damian-garrido Any update on this one?

**Comment by damian-garrido** [comment]
@MJWainwright 
i have some improvements about this, check the video

https://github.com/user-attachments/assets/1a704b7c-6727-4624-8632-4175772141b8

as you can see, now the text related to the options is displayed above the textarea and don't delete the text that the user put in when you change your option.
also i added a lighter bold to the explanation text, because this is missing when you want to read the explanation.

but also, i have a doubt related to the GRADE-CERQual assessment of confidence, in this section, the text should be displayed as the other? for the moment i don't display nothing as you can see in the following video

https://github.com/user-attachments/assets/d2e9ba08-830c-4277-a45f-eec675442798

let me know anything!

**Comment by damian-garrido** [comment]
@MJWainwright could you check this?

**Comment by MJWainwright** [comment]
@damian-garrido Apologies all my iSoQ time was put towards preparing for the webinar at the end of the month. I'm now going to be off work for 2 weeks. @javierapena could you check the update Damián has made and provide feedback to him please? I will check when I return as well. Thanks!

**Comment by javierapena** [comment]
Hi Megan!
Of course, we can check it with Damian and let you know :)

**Comment by MJWainwright** [comment]
Checked on test server, ready to publish to production server

**Comment by MJWainwright** [comment]
@damian-garrido I do not see the change you showed me yesterday on the test server. I cleared my cache. It just looks exactly like it did before (a notes box, and an explanation box that appears only when you make a selection and has the starting text in the box rather than above it)
 You said develop server on your email, is that the same as test server? The real site is the production server, correct?

**Comment by damian-garrido** [comment]
@MJWainwright sorry!
my mistake, i publish on the test server, please check again!

**Comment by damian-garrido** [comment]
@MJWainwright i found an issue related to the new approach for the preventing delete content.
now, with the actual implementation, we missing the functionality of the "GRADE-CERQual assessment of confidence", the content does not contain the pre-filled data according to the other responses.
but also, if i have complete those steps but then i replace one of them to another option, the "GRADE-CERQual assessment of confidence" did not change those values to the new one.
what do you think is the right way to do this?

**Comment by MJWainwright** [comment]
@damian-garrido I discovered an issue. The start of the sentence in bold is missing when the user selects no/very minor concerns. (see below). The text "No/very minor concerns regarding methodological limitations because" needs to appear above the box. HOWEVER, for no/very minor the explanation is optional, so if the user does not enter any text in the box then the text "no/very minor concerns regarding methodological limitations because..." should NOT appear in the evidence profile when the user clikcs save. 
![image](https://github.com/user-attachments/assets/afc57071-3066-4e8c-a794-947aed4ce13a)

**Comment by MJWainwright** [comment]
@damian-garrido Another error is that the warning text "it looks like you have not finished writing an explanation" modal appears when the user selects no/very minor. This is incorrect. As an explanation is not required for "No/very minor" this modal should not appear if the user does not write an explanation. Look at the production server, it is correct there. 
![image](https://github.com/user-attachments/assets/860aa8e6-87e7-4d90-9b67-3e120096f143)

**Comment by MJWainwright** [comment]
@damian-garrido Another thing that is on the production server and is now missing on the test server is the minimum text in the explanation. This should be populated automatically in the box. See the problem in the test server in the first image, and then see the correct functionality in the production server in the second photo

![image](https://github.com/user-attachments/assets/9f224ed9-c637-41d3-902d-422f5e9edd79)
![image](https://github.com/user-attachments/assets/321c474c-d532-4750-a11e-4eb7b29d36ef)

**Comment by damian-garrido** [comment]
@MJWainwright i made the changes for bringing back the pre-populated explanation for the cerqual, also remove the modal for require an explanation in case you select "No/very minor concerns", also "No/very minor concerns" appears in bold now.
the option for change the cerqual content when you change your mind in the other options is not possible, because this action will remove the whole content.
i could add in bold the pre-populated content for cerqual, like the others, and leave with no changes the coment of the users. this option will work for you?

**Comment by MJWainwright** [comment]
@damian-garrido 
checked pre-pop text for overall assessment - all good
checked that requirement to add an explanation for no/very minor was removed - all good

But, the issue is, and perhaps that's been this way and we've not seen it before, but if the user completes all 4 assessments and then completes the overall assessment, and then goes back to one of the components and chnages an assessment, the overall assessment explanation text remains the same and is in fact incorrect (e,g, if at first I selected minor concerns for coherence, do the overall assessment, then co back to coherence and change to serious concerns, the pre-populated text in the explanation for the overall assessment still says "minor concerns". So this is a problem. I'm thinking that if a user changes a component assessment, then they should have to redo their overall assessment, so perhaps the action of changing a level of concern for any of the components should delete the overall assessment and revert back to that column saying "assessment not completed". But the trade-off is that the user will loose any of the content they added to the prepopulated explanation, and would loose the content they've written into the notes box. Maybe we need a call to discuss this.

**Comment by damian-garrido** [comment]
@MJWainwright @javierapena 
one option will be show a modal warning (when the cerqual is completed) to the user that say if they change their option, the grade-cerqual will be removed, so, the user need to copy or make a backup of the content and then come back to change their option and repopulate the grade-cerqual response.
what do you think?

**Comment by damian-garrido** [comment]
@MJWainwright no, actually it's possible to keep the whole content. (selected option, explanation and notes)
i'm try to say that we can add a modal before the user save the new selected option because remove the content from grade-cerqual without tell to the user will be a little odd, no?
at least they could save some data from the explanation before this will be deleted.

**Comment by MJWainwright** [comment]
@damian-garrido . Sorry I'm not understanding you. Maybe we have a quick meeting about this. I'll email you.

**Comment by damian-garrido** [comment]
@MJWainwright sure, today i could, also tomorrow.

**Comment by MJWainwright** [comment]
@damian-garrido Tomorrow would be great

**Comment by damian-garrido** [comment]
@MJWainwright i have an update of this

**Comment by MJWainwright** [comment]
@damian-garrido There's a problem! I've started a new project, and the warning modal is appearing as soon as I click a radio button. For example, in the new project below, I clicked methodological limitations, then "moderate concerns" and the warning appears. This is wrong. The warning needs to appear only if the Overall Assessment column (column 5) is complete and then the user changes a radio button choice in any of the components. Can you please revise?

![Image](https://github.com/user-attachments/assets/0a8f1453-3a4b-446f-81c7-b4131a43523c)

**Comment by MJWainwright** [comment]
@damian-garrido . There's an issue. When I complete all five assessessments and then go back to one of the four components (e.g. methodological limitations) and choose a different radio button the warning comes up and if I click "YES" then the text in the explanation box disappears. This is incorrect. Clicking "yes" should only remove the explanation text from the overall assessment (5th column), not the individual component I am working on. Can you please fix this for all four components?

The second issue I saw was the page jump. Everytime I complete one of the five assessments the page jumps to the top and I have to scroll down to the Evidence Profile again. 

![Image](https://github.com/user-attachments/assets/aa87744c-8108-40ab-afed-348c0061ca06)

![Image](https://github.com/user-attachments/assets/69886afc-be98-47ec-a30d-fd19a113994a)

![Image](https://github.com/user-attachments/assets/c0b2c9d8-53a7-472a-9238-6e45d16217f7)

**Comment by damian-garrido** [comment]
@MJWainwright i made an update of this. could you check if the changes fits the requirements?

**Comment by MJWainwright** [comment]
@damian-garrido checked on production server. All good!
